### PR TITLE
fix(new-dashboard): team member invite & creation

### PR DESCRIPTION
**Pull request by jyash97**
## What kind of change does this PR introduce?
Bug Fix

## What is the current behavior?
1. We can invite ourself in the workspace -> which throws an error
https://www.loom.com/share/d8457a2f5d684f8899712301f4f37951
2. On creating workspace -> empty workspaces crashes new dashboard

## What is the new behavior?
https://www.loom.com/share/d4144fb1fb3647208903733b79a209ab

## What steps did you take to test this? This is required before we can merge, make sure to test the flow you've updated.

1. Tested the member autocomplete
2. Creation of the empty workspace

## Checklist

- [x] Testing 
- [x] Ready to be merged
      
- [x] Added myself to contributors table

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/4487/commits/4a99eaeb41ce6f742f95385926aed566ce20e944 is at https://pr4487.build.csb.dev/s/new.

File: packages/app/src/app/components/UserSearchInput/UserSearchInput.tsx [hunk 1]
```diff
@@ -11,17 +12,27 @@ type User = {
...
+  inputValue,
+  children,
+  allowSelf,
+}: IUserAutoComplete) => {
   const [users, setUsers] = React.useState<User[]>([]);
   const [loading, setLoading] = React.useState<boolean>(true);
   const [error, setError] = React.useState<Error | null>(null);
+  const {
+    state: {
+      user: { username },
```
Comment (CompuIves) [inline packages/app/src/app/components/UserSearchInput/UserSearchInput.tsx:33]: I think you should also check if user is there, sometimes people will use this component while not signed in.

File: packages/app/src/app/components/UserSearchInput/UserSearchInput.tsx [hunk 2]
```diff
@@ -50,14 +64,15 @@ const UserAutoComplete = ({ inputValue, children }: IUserAutoComplete) => {
...
-  }, [inputValue]);
+  }, [allowSelf, inputValue, username]);
 
   return children({ users, loading, error });
 };
 
 interface IUserSearchInputProps {
   onInputValueChange: DownshiftProps<string>['onInputValueChange'];
   inputValue: string;
+  allowSelf: boolean;
```
Comment (CompuIves) [inline packages/app/src/app/components/UserSearchInput/UserSearchInput.tsx:75]:
```suggestion
  allowSelf?: boolean;
```

File: packages/app/src/app/pages/NewDashboard/Content/routes/Settings/NewTeam.tsx [hunk 3]
```diff
@@ -17,14 +17,26 @@ export const NewTeam = () => {
...
     const teamName = event.target.name.value;
-    console.warn(teamName, 'created by', user.username);
-    setLoading(true);
-    try {
-      await dashboard.createTeam({ teamName });
-      setLoading(false);
-      history.push(dashboardUrls.teamInvite(activeTeam));
-    } catch {
-      setLoading(false);
+    if (teamName && teamName.trim()) {
```
Comment (CompuIves) [inline packages/app/src/app/pages/NewDashboard/Content/routes/Settings/NewTeam.tsx:20]: Are changes needed here too?

File: packages/app/src/app/components/UserSearchInput/UserSearchInput.tsx (hunk 2 above)
Comment (CompuIves) [reply packages/app/src/app/components/UserSearchInput/UserSearchInput.tsx:75]: Let's make it optional! I think we always want you to not see yourself, haha

File: packages/app/src/app/components/UserSearchInput/UserSearchInput.tsx [hunk 4]
```diff
@@ -67,6 +82,7 @@ const InputWithoutTypes = Input as any;
 export const UserSearchInput = ({
   onInputValueChange,
   inputValue,
+  allowSelf = true,
```
Comment (MichaelDeBoey) [inline packages/app/src/app/components/UserSearchInput/UserSearchInput.tsx:85]:
```suggestion
  allowSelf = false,
```

File: packages/app/src/app/components/UserSearchInput/UserSearchInput.tsx [hunk 5]
```diff
@@ -11,17 +12,27 @@ type User = {
 
 interface IUserAutoComplete {
   inputValue: string;
+  allowSelf: boolean;
```
Comment (MichaelDeBoey) [inline packages/app/src/app/components/UserSearchInput/UserSearchInput.tsx:15]:
```suggestion
  allowSelf?: boolean;
```

File: packages/app/src/app/components/UserSearchInput/UserSearchInput.tsx [hunk 6]
```diff
@@ -11,17 +12,27 @@ type User = {
...
     loading: boolean;
     error: Error | null;
   }) => JSX.Element;
 }
 
-const UserAutoComplete = ({ inputValue, children }: IUserAutoComplete) => {
+const UserAutoComplete = ({
+  inputValue,
+  children,
+  allowSelf,
```
Comment (MichaelDeBoey) [inline packages/app/src/app/components/UserSearchInput/UserSearchInput.tsx:26]:
```suggestion
  allowSelf = false,
```

File: packages/app/src/app/components/UserSearchInput/UserSearchInput.tsx (hunk 1 above)
Comment (jyash97) [reply packages/app/src/app/components/UserSearchInput/UserSearchInput.tsx:33]: Fixed 👍

File: packages/app/src/app/pages/NewDashboard/Content/routes/Settings/NewTeam.tsx (hunk 3 above)
Comment (jyash97) [reply packages/app/src/app/pages/NewDashboard/Content/routes/Settings/NewTeam.tsx:20]: What changes are you referring to here ?

**Comment by jyash97** [comment]
Thanks @CompuIves  & @MichaelDeBoey  for the review. Have applied the changes

**Comment by SaraVieira** [comment]
Hey!


Tested and works great! Thank you for the fix and for the time to do the PR changes :)

**Comment by jyash97** [comment]
Hey @SaraVieira  Thanks for the review and testing 👍 

Great work on the new dashboard 💯 . Waiting for the new dashboard to go live 🚀

**Pull request merged**